world_boss/app/raid.py

```diff
@@ -262,3 +268,235 @@ def get_jwt_auth_header() -> dict[str, str]:
 
 def get_tx_delay_factor(index: int) -> int:
     return 4 * (index // 4)
+
+
+def get_transfer_assets_plain_value(
+    sender: str, recipients: typing.List[Recipient], memo: str
+) -> ActionPlainValue:
+    values: TransferAssetsValues = {
+        "sender": bytes.fromhex(sender.replace("0x", "")),
+        "recipients": [],
+    }
+    for r in recipients:
+        amount = r["amount"]
+        decimal_places = amount["decimalPlaces"]
+        values["recipients"].append(
+            [
+                bytes.fromhex(r["recipient"].replace("0x", "")),
+                [
+                    {
+                        "decimalPlaces": decimal_places.to_bytes(1, "big"),
+                        "minters": None,
+                        "ticker": amount["ticker"],
+                    },
+                    amount["quantity"] * 10**decimal_places,
+                ],
+            ]
+        )
+    if memo is not None:
+        values["memo"] = memo
+    pv: ActionPlainValue = {
+        "type_id": "transfer_assets3",
+        "values": values,
+    }
+    return pv
+
+
+def create_unsigned_tx(
+    planet_id: str,
+    public_key: bytes,
+    address: str,
+    nonce: int,
+    plain_value: ActionPlainValue,
+    timestamp: datetime.datetime,
+) -> bytes:
+    """
+    :param planet_id: planet id for genesis hash
+    :param public_key: signer public key
+    :param address: signer address
+    :param nonce: tx nonce
+    :param plain_value: tx actions plain value
+    :param timestamp: tx time stamp
+    :return: bencoded unsigned tx
+    """
+    if address.startswith("0x"):
+        address = address[2:]
+    return bencodex.dumps(
+        {
+            # Raw action value
+            b"a": [plain_value],
+            # Genesis block hash
+            b"g": get_genesis_block_hash(planet_id),
+            # GasLimit (see also GasLimit list section below)
+            b"l": 4,
+            # MaxGasPrice (see also Mead section for the currency spec)
+            b"m": [
+                {"decimalPlaces": b"\x12", "minters": None, "ticker": "Mead"},
+                1000000000000000000,
+            ],
+            # Nonce
+            b"n": nonce,
+            # Public key
+            b"p": public_key,
+            # Signer
+            b"s": bytes.fromhex(address),
+            # Timestamp
+            b"t": timestamp.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+            # Updated addresses
+            b"u": [],
+        }
+    )
+
+
+def append_signature_to_unsigned_tx(unsigned_tx: bytes, signature: bytes) -> bytes:
```

Comment:
I think some tx related functions are duplicated with thing in `kms.py`.
Is it intended?